security/pkg/nodeagent/caclient/interface/iclient.go

```diff
@@ -20,6 +20,13 @@ import (
 
 // Client interface defines the clients need to implement to talk to CA for CSR.
 type Client interface {
+	// if withToken set to true the CSR request will attach the token , otherwise it will not
 	CSRSign(ctx context.Context, reqID string, csrPEM []byte, subjectID string,
-		certValidTTLInSec int64) ([]string /*PEM-encoded certificate chain*/, error)
+		certValidTTLInSec int64, withToken bool) ([]string /*PEM-encoded certificate chain*/, error)
+	/* if the isRotate is set to True, which means it is reconnected special for rotate cert case
```

Comment:
That's a pretty strange interface. Too specific - this is an implementation detail, not a function that should be present in all clients signing certificates. 

We're not adding one interface method for each small behavior change.

Reply:
ok no problem yes I think it should be deleted. and refactor